### PR TITLE
Add dagger hilt

### LINKED ISSUE #102: Add Dagger Hilt

### PR DESCRIPTION
**Pull request by Irineu333**
### Description
I improved the app's dependency injection with dagger hilt.

### CODE REVIEW

**Review by PatrykMis — APPROVED** [review]

**Comment by PatrykMis** [comment]
To be consistent, I added 2 more commits to refactor for Version Catalogs.

Slight offtopic, but buildSrc modules are problematic, thus I had to do a workarround from https://github.com/gradle/gradle/issues/20084. I think that build logic could be safely moved to app module build file. For example, look at https://github.com/chenxiaolong/BCR/blob/master/app/build.gradle.kts

File: build.gradle.kts [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
 plugins {
-    id("com.google.dagger.hilt.android") version "2.48" apply false
-    id("com.google.devtools.ksp") version "1.9.20-1.0.14" apply false
```
Comment (Irineu333) [inline build.gradle.kts]: Thanks, I would refactor this today.

**Comment by Irineu333** [comment]
> To be consistent, I added 2 more commits to refactor for Version Catalogs.
> 
> Slight offtopic, but buildSrc modules are problematic, thus I had to do a workarround from [gradle/gradle#20084](https://github.com/gradle/gradle/issues/20084). I think that build logic could be safely moved to app module build file. For example, look at https://github.com/chenxiaolong/BCR/blob/master/app/build.gradle.kts

I am aware of this limitation of `plugins.alias` with `buildSrc`, I believe your solution is ideal. Let's hope they improve this.

**Comment by Irineu333** [comment]
@PatrykMis I sent a commit improving context dependencies.

**Review by PatrykMis — APPROVED** [review]
For me as a beginner, it's probably too early to understand benefits of using Hilt, dependency injection and this topic at all. Approving with trust you did (like always) the great job. Tested on my devices and it works great and fast.

**Comment by Irineu333** [comment]
Last change, I swear.

**Review by PatrykMis — APPROVED** [review]

File: app/src/main/java/com/neo/speaktouch/utils/NodeValidator.kt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
- * Node focus and read validations.
+ * [FILE DESCRIPTION].
  *
- * Copyright (C) 2023 Irineu A. Silva.
+ * Copyright (C) 2023 [AUTHOR].
```
Comment (Irineu333) [inline app/src/main/java/com/neo/speaktouch/utils/NodeValidator.kt:4]: I saw this just now. Something like this couldn't have been overlooked. This class is one of, if not the most important in the app

Comment (Irineu333) [inline app/src/main/java/com/neo/speaktouch/utils/NodeValidator.kt:4]: I will correct it directly in develop.

**Pull request merged**